src/azure-cli-core/azure/cli/core/_help.py

```diff
@@ -192,8 +195,8 @@ def update_loaders_with_help_file_contents(self, nouns):
 
     # This method is meant to be a hook that can be overridden by an extension or module.
     @staticmethod
-    def example_provider(help_file):
-        return help_file.examples
+    def check_for_new_examples(command):  # pylint: disable=unused-argument
+        return []
```

Comment:
My concern is by taking `command` as argument and use `self.examples = new_examples` later, this will totally remove the built-in examples. This is really intended?

Reply:
So the current plan is to entirely replace the default examples, but I have debated how to best build this hook. 

I like the idea of leaving it extensible, so in the future the extension could do some form of combination without needing to update the core CLI again. I'll make a change that will pass in the default examples, which should also simplify the logic here too.

Reply:
The change is great!